src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/BookingExtractionService.kt

```diff
@@ -192,14 +193,30 @@ class BookingExtractionService(
     if (mostRecentSentencesByReleaseDate.any { it.isRecall() }) {
       dates[PRRD] = latestReleaseDate
     }
-    if (booking.sentences.all { it is DetentionAndTrainingOrderSentence }) {
+    if (booking.sentences.any { it is DetentionAndTrainingOrderSentence }) {
       dates[MTD] = latestReleaseDate
-      if (mostRecentSentenceByExpiryDate.releaseDateTypes.contains(ETD)) {
-        calculateEtd(mostRecentSentenceByExpiryDate, dates)
-      }
-
-      if (mostRecentSentenceByExpiryDate.releaseDateTypes.contains(LTD)) {
-        calculateLtd(mostRecentSentenceByExpiryDate, dates)
+      calculateWhenAllDtos(mostRecentSentenceByExpiryDate, dates)
+      if (sentences.size > 1 && !sentences.any { it is ConsecutiveSentence } && !sentences.all { it is DetentionAndTrainingOrderSentence }) {
+        val firstNonDtoSentence = sentences.first { it !is DetentionAndTrainingOrderSentence }
```

Comment:
Why is this just first? What is the list sorted by? This may work if there is only one concurrent non dto, but what happens if there is more?

i think generally we need a better way of checking for DTOs.

Something like adding a function to `CalculableSentence.isDto()`. Then implement it for Consecutive sentence, for single term sentences and DTO sentences.

Otherwise we'll need to do this check everywhere

`it is DetentionAndTrainingOrderSentence || (it is ConsecutiveSentence && it.orderedSentences.all{ it is DetentionAndTrainingOrderSentence }) || it is DtoSingleTermSentence`

Reply:
Done.